clippy_lints/src/methods/mod.rs

```diff
@@ -1975,6 +1975,19 @@ fn lint_search_is_some<'a, 'tcx>(
         );
         let search_snippet = snippet(cx, search_args[1].span, "..");
         if search_snippet.lines().count() <= 1 {
+            // suggest `any(|x| ..)` instead of `any(|&x| ..)` for `find(|&x| ..).is_some()`
+            let any_search_snippet =
+                if_chain! {
+                    if search_method == "find";
+                    if let hir::ExprKind::Closure(_, _, body_id, ..) = search_args[1].node;
```

Comment:
This may panic for `find` functions with 0 or 1 argument. I don't really know if this is possible at this point, but better save than sorry:
```suggestion
                    if let Some(arg) = search_args.get(1);
                    if let hir::ExprKind::Closure(_, _, body_id, ..) = arg.node;
```

Reply:
Thank you 😸

We can ensure the caller of search is an Iterator, so the `search_args[1]` should be fine:

https://github.com/rust-lang/rust-clippy/blob/c3fde34fd53a6eed89dd30064b1985acf39f7f07/clippy_lints/src/methods/mod.rs#L1969-L1977